sensor/common/clusterentities/debug.go

```diff
@@ -30,10 +30,10 @@ func (e *Store) Debug() []byte {
 	m["endpoints"] = e.endpointsStore.debug()
 	m["IPs"] = e.podIPsStore.debug()
 	m["containerIDs"] = e.containerIDsStore.debug()
-	// json pretty-printer will sort it for us
-	concurrency.WithRLock(&e.traceMutex, func() {
-		m["events"] = e.trace
-	})
+	// json pretty-printer will sort it for us.
+	// We need to copy the trace map, otherwise json.Marshal might panic when
+	// reading the map if track is called at the same time.
+	m["events"] = e.getTrace()
```

Comment:
Let's maybe use https://pkg.go.dev/maps#Clone as it does exactly this

Reply:
Nice! I didn't know that function existed.